cdisc_rules_engine/services/data_services/local_data_service.py

```diff
@@ -168,29 +177,37 @@ def _async_get_datasets(
             )
 
     def read_metadata(self, file_path: str) -> dict:
+        # print("local_data_service.py: reading metadata from file = ", file_path)
         file_size = os.path.getsize(file_path)
         file_metadata = {
             "path": file_path,
             "name": extract_file_name_from_path_string(file_path),
             "size": file_size,
         }
         with open(file_path, "rb") as f:
-            contents_metadata = DatasetMetadataReader(f.read()).read()
+            if not self.use_json:
+                contents_metadata = DatasetMetadataReader(f.read()).read()
+            # 2022-11-18: FOR USE WITH Dataset-JSON only
+            else:
+                contents_metadata = DatasetJSONMetadataReader(file_path).read()
```

Comment:
We should be able to detect whether or not it is a json file or not based on the file extension